lstchain/high_level/interpolate.py

```diff
@@ -0,0 +1,577 @@
+import numpy as np
+import logging
+import astropy.units as u
+from astropy.table import QTable, Table
+from astropy.io import fits
+from astropy.time import Time
+
+from lstchain.__init__ import __version__
+
+from pyirf.io.gadf import (
+    create_aeff2d_hdu,
+    create_energy_dispersion_hdu,
+    create_psf_table_hdu,
+)
+from pyirf.interpolation import (
+    interpolate_effective_area_per_energy_and_fov,
+    interpolate_energy_dispersion,
+    interpolate_psf_table,
+    # interpolate_rad_max,
+)
+from scipy.spatial import Delaunay, distance
+from scipy.interpolate import griddata
+
+log = logging.getLogger(__name__)
+
+
+def interp_params(params_list, data):
+    """
+    From a given list of angular parameters, to be used for interpolation,
+    take values from a given data table/dict.
+
+    Returns the neccessary values with applied functions as need be for the
+    interpolation, for each parameter as a list.
+    """
+    mc_pars = []
+    if "ZEN_PNT" in params_list:
+        mc_pars.append(np.cos(u.Quantity(data["ZEN_PNT"], "deg").to_value(u.rad)))
```

Comment:
Then the unit should be read instead of hardcoding "deg"

Reply:
Yes, this was part of an earlier discussion with Max, and I just fixed the angular values to be in degrees for IRF HDU metadata and the `data_params` read in `lstchain/io/io/read_data_dl2_to_QTable` for real data's target information, so as to have the values in good human-readable units and use radian units when performing any trigonometric computation.

Further generalization seems irrelevant to our use-case, but if anyone has a strong opinion on this, it can be done.

Reply:
Ok as is for me